### PR TITLE
Add in lines 15 and 16 to MD Form 502

### PR DESCRIPTION
**Pull request by anisharamnani**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/ABCD-123](https://codeforamerica.atlassian.net/browse/FYST-655
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
Lines 15 and 16 for MD Form 502 were added. Lines 9, 10a, and 11 were stubbed. 
- Alternatives considered
## How to test?
* the DF data for "Todd 1099 int" works 
* you should see lines 15 and 16 filled in 
Screenshot 2024-11-14 at 4 27 45 PM

### CODE REVIEW

File: app/lib/efile/md/md502_calculator.rb [hunk 1]
```diff
@@ -361,12 +376,12 @@ def calculate_deduction_method
         s_mfs_d: {
           12000 => 1_800,
           17999 => ->(x) { x * 0.15 },
-          18000 => 2_700,
+          Float::INFINITY => 2_700,
```
Comment (anisharamnani) [inline app/lib/efile/md/md502_calculator.rb:379]: @jenny-heath helped sort out there was a bug here where its supposed to be "$18,000 or over" and so the band is changed in this line. Here is the original [ticket](https://codeforamerica.atlassian.net/browse/FYST-758) for reference.

File: spec/lib/efile/md/md502_calculator_spec.rb [hunk 2]
```diff
@@ -818,18 +818,42 @@
...
+  end
+
   describe "#calculate_line_17" do
     context "when method is standard" do
       [
         [["single", "married_filing_separately", "dependent"], [
           [12_000, 1_800],
           [17_999, 17_999 * 0.15],
-          [18_000, 2_700],
+          [18_001, 2_700],
```
Comment (anisharamnani) [inline spec/lib/efile/md/md502_calculator_spec.rb:851]: changed the tests to expose the above mentioned [bug](https://github.com/codeforamerica/vita-min/pull/4980/files#r1842913316).

File: spec/lib/pdf_filler/md502_pdf_spec.rb [hunk 3]
```diff
@@ -440,5 +439,21 @@
         expect(pdf_fields["Enter 13"].to_i).to eq 0
       end
     end
+
+    describe "Line 15" do
```
Comment (jenny-heath) [inline spec/lib/pdf_filler/md502_pdf_spec.rb:443]: minor thing but there's a section above `context "subtractions" do` that you could move these blocks into

**Review by jenny-heath — APPROVED** [review]
looks great!

**Pull request merged**